app/client/lib/ethereum/observeEvents.js

```diff
@@ -12,15 +14,31 @@ observeEvents = function(){
     */
     collectionObservers[collectionObservers.length] = Events.find({}).observe({
         /**
-        This will observe the transactions creation and create watchers for outgoing transactions, to see when they are mined.
+        This will observe when events are added and link it to the custom contract.
 
         @method added
         */
         added: function(newDocument) {
+            
+            // This creates a temporary cache for the contracts, to reduce the amount of db reads            
```

Comment:
I believe that you are trying to reduce the amount of writes/updates instead of reads?